### PR TITLE
Paste tooltip

### LINKED ISSUE #179: Paste tooltip should show what book is currently saved
Only that means we'll need to instrument tooltips such that they can be changed after the fact

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #179 

Kapture 2023-01-06 at 00 48 35

should merge #231 and #234 first before reviewing this

### CODE REVIEW

File: src/ts/content/extensions/copy/onEdit.ts [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+	onceFormRender,
+	onFormRemoved,
+	onFormRender,
+} from "../../entities/bookForm";
 import {showToast, ToastType} from "../../../common/ui/toast";
-import {appendRow, saveFormData} from "./common";
+import {appendRow, makeButton, saveFormData} from "./common";
+import {addTooltip} from "../../../common/ui/tooltip";
+import {onBackgroundEvent} from "../../util/onBackgroundEvent";
+import {BackgroundEvent} from "../../../common/backgroundEvent";
```
Comment (adirar111) [inline src/ts/content/extensions/copy/onEdit.ts:14]: github action warnings seems legit when i checkout locally

File: src/ts/common/ui/tooltip.ts [hunk 2]
```diff
@@ -15,22 +15,28 @@ interface TooltipOptions {
...
 }
 
-const tooltipped = <T extends HTMLElement>(
-	element: T,
+const tooltipped = <T extends HTMLElement>(element: T, options: TooltipOptions): T => {
+	addTooltip(element, options);
+	return element;
+};
+
+const addTooltip = (
```
Comment (adirar111) [inline src/ts/common/ui/tooltip.ts:23]: ah we're separating this out of tooltipped to support this feature?

**Review by adirar111 — APPROVED** [review]
super cool! def a better ux

File: src/ts/content/extensions/copy/onEdit.ts (hunk 1 above)
Comment (braxtonhall) [reply src/ts/content/extensions/copy/onEdit.ts:14]: OOPS

**Pull request merged**